observation_portal/common/state_changes.py

```diff
@@ -161,8 +157,10 @@ def debit_ipp_time(request_group):
         total_duration_dict = request_group.total_duration
         for tak, duration in total_duration_dict.items():
             duration_hours = duration / 3600
-            time_allocations_dict[tak].ipp_time_available -= (ipp_value * duration_hours)
-            time_allocations_dict[tak].save()
+            ipp_difference = ipp_value * duration_hours
+            with transaction.atomic():
+                TimeAllocation.objects.select_for_update(of=('self')).filter(id=time_allocations_dict[tak].id).update(
```

Comment:
I'm not 100% sure the of=('self') does anything meaningful yet, but I am trying to avoid it locking related tables to the TimeAllocation (like proposal/semester). I saw evidence of those being locked before when looking through the dev database locks during deadlocks

Reply:
Did you decide anything on this? Seems like it couldn't hurt but yeah, dunno exactly.

Reply:
yeah I tested this out and it seems it is not necessary. After reading the docs a few more times, I think it only locks other rows and needs this if you do a 'select_related' on the queryset